test/replication.test.js

```diff
@@ -52,6 +52,121 @@ describe('Replication / Change APIs', function() {
     };
   });
 
+  describe('optimization check rectifyChange Vs rectifyAllChanges', function() {
+    beforeEach(function initialData(done) {
+      var data = [{name: 'foo1', lname: 'bar1'}, {name: 'foo2', lname: 'bar2'}];
+      async.auto({
+        createSourceModel: function(callback) {
+          SourceModel.create(data, callback);
+        },
+        replicate: ['createSourceModel', function(callback, results) {
+            SourceModel.replicate(TargetModel, callback);
+          }]
+      }, function(err, results) {
+        done(err);
+      });
+    });
+
+    it('rectifyOnDelete for Delete should call rectifyChange instead of rectifyAllChanges', function(done) {
+      var count = 0;
+      TargetModel.rectifyChange = function() {
+        // based on destroyAll where condition it should call rectifyChange two times
+        if (++count == 2) {
+          TargetModel.find(function(err, data) {
+            expect(data.length).to.equal(0);
+            return done(err);
+          });
+        }
+      };
+      TargetModel.rectifyAllChanges = function() {
+        return done(new Error('Should not call rectifyAllChanges'));
+      };
+
+      async.auto({
+        destroySourceModel: function(callback) {
+          SourceModel.destroyAll({'or': [{name: 'foo1'}, {lname: 'bar2'}]}, callback);
```

Comment:
IIRC, the current implementation does not support `or` in the filters. Why does this query trigger two `rectifyChange` calls? Also `name` and `lname` do not look like id properties. If that's true, then `rectifyAllChange` should have been triggered instead. We cannot run `rectifyChange` when we don't know the exact model id.